libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -1003,11 +1003,24 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
             response.statusCode = StatusCodes.BAD_REQUEST;
             response.setBody(`Request missing verb and/or path. Verb: ${ request.verb }. Path: ${ request.path }`);
             return response;
-        } 
+        }
 
         if (request.verb.toLocaleUpperCase() === GET && request.path.toLocaleLowerCase() === VERSION_PATH) {
             response.statusCode = StatusCodes.OK;
-            response.setBody({UserAgent: USER_AGENT});
+
+            if (!this.credentials.appId || !this.credentials.appPassword) {
```

Comment:
Look at my side PR: https://github.com/microsoft/botbuilder-js/pull/1337

I would love to not condition things on the existance of the password because we have to assume that not all auth methods use passwords. This won't even compile when I merge my PR. Can we rely on app id + the call to Gettoken() in line 1017?

Reply:
Done!